lib/archethic/self_repair/sync/transaction_handler.ex

```diff
@@ -35,18 +35,44 @@ defmodule Archethic.SelfRepair.Sync.TransactionHandler do
           transaction_summary: %TransactionSummary{
             address: address,
             type: type,
-            movements_addresses: mvt_addresses
+            movements_addresses: mvt_addresses,
+            genesis_address: genesis_address,
+            version: summary_version
           }
         },
         node_list
       ) do
-    chain_storage_nodes = Election.chain_storage_nodes_with_type(address, type, node_list)
+    last_chain_storage_nodes = Election.chain_storage_nodes_with_type(address, type, node_list)
+    genesis_chain_storage_nodes = Election.chain_storage_nodes(genesis_address, node_list)
+    node_key = Crypto.first_node_public_key()
 
-    if Utils.key_in_node_list?(chain_storage_nodes, Crypto.first_node_public_key()) do
+    if Utils.key_in_node_list?(last_chain_storage_nodes, node_key) or
+         Utils.key_in_node_list?(genesis_chain_storage_nodes, node_key) do
       not TransactionChain.transaction_exists?(address)
     else
       io_node? =
```

Comment:
To optimize the request, you should previously ensure the transaction does not already exist it the DB to avoid doing IO and having a "transaction already exists" after